src/tools/schema2kotlin.ts

```diff
@@ -205,8 +252,26 @@ Schema(
         singletons = ${leftPad(ktUtils.mapOf(this.singletonSchemaFields, 30), 8, true)},
         collections = ${leftPad(ktUtils.mapOf(this.collectionSchemaFields, 30), 8, true)}
     ),
-    "${schemaHash}"
-)`;
+    "${schemaHash}",
+    refinement = ${this.refinement},
+    query = ${this.query}
+  )`;
+  }
+
+  typeFor(name: string): string {
+    return typeFor(name);
+  }
+
+  generatePredicates() {
+    const expression = KTExtracter.fromSchema(this.node.schema, this);
+    const refinement = this.node.schema.refinement;
+    const queryType = refinement.getQueryParams().get('?');
+    const lines = expression.split('\n').join('\n        ');
+    if (queryType) {
+      this.query = `{ data, queryArgs ->\n        ${lines}\n    }`;
```

Comment:
Now that I see this a bunch, it think it'd be worth while to add a `lambda` application creator to `kotlin-generation-utils.ts`.

Reply:
That'd be really helpful but is out of scope for this PR. Feel free to assign a bug to to me as future work if you don't have cycles for it. :)